lib/factory.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+class Factory
+  def self.new(*arguments, &block)
+    if arguments.first.is_a?(String)
+      raise NameError unless arguments.first.match?(/^[A-Z]/)
+      class_name = arguments.first
```

Comment:
What about `Array.shift`?

Reply:
'shift' removes the first element and returns it. 'first' just returns the first element without removing. Removing the first element of the array is an expensive operation and is not required here

Reply:
How expensive is it? It is so expensive that you are going to write 3 lines of code instead of 1?

Reply:
Ruby is slow by its nature, nevertheless developers prefer using Ruby instead of Assembler or C++. Because Ruby is beautiful and concise programming language. Plenty of things you can do in one line.